external/webserver/http_client.c

```diff
@@ -476,6 +477,30 @@ int http_parse_message(char *buf, int buf_len, int *method, char *url,
 						http_dispatch_url(client, req);
 						process_finish = true;
 						read_finish = true;
+
+						/* Process the Trailer */
+						header_field_value = http_keyvalue_list_find(params, "Trailer");
+						if (strncmp(header_field_value, "(null)", 6)) {
+							len->sentence_start = sentence_end + 2;
+
+							/* match the Trailer field */
+							if (strncmp(buf + len->sentence_start, header_field_value, strlen(header_field_value)) == 0) {
+								HTTP_LOGE("Trailer header: %s\n", header_field_value);
+								sentence_end = http_find_first_crlf(buf, buf_len, len->sentence_start);
+								if (sentence_end < 0) {
+									HTTP_LOGE("Error: Not accord with chunked encoding\n");
+									read_finish = true;
+									return read_finish;
+								}
+
+								/* copy Trailer data */
+								memset(entity, 0, HTTP_CONF_MAX_ENTITY_LENGTH);
```

Comment:
is it guaranteed that the size of entity is less than len->sentence_end - len->sentence_start?

Reply:
done